### PR TITLE
don't force-repoll on implicit team chat send hot path

### PR DESCRIPTION
**Pull request by maxtaco**
- fix the test that this broke, and in general:
- fix the notification system internal to the app that adjusts the locally cached version of the teamchain
 - sometimes bump the chain 0, sometimes 1, sometimes 2, depending on the caller!
 - previously, always bumping +1 definitely caused some bugs, but we likely got away with it

### CODE REVIEW

**Comment by maxtaco** [comment]
cc @mmaxim  this should fix the chat perf bug

File: go/teams/teams.go [hunk 1]
```diff
@@ -666,10 +669,17 @@ func (t *Team) Leave(ctx context.Context, permanent bool) error {
...
+	if err != nil {
+		return err
+	}
+
+	t.notify(ctx, keybase1.TeamChangeSet{MembershipChanged: true}, latestSeqno)
+	return nil
 }
 
 func (t *Team) deleteRoot(ctx context.Context, ui keybase1.TeamsUiInterface) error {
+	m := t.MetaContext(ctx)
```
Comment (mlsteele) [inline go/teams/teams.go]: mctx

File: go/teams/transactions.go [hunk 2]
```diff
@@ -836,7 +836,7 @@ func (tx *AddMemberTx) Post(mctx libkb.MetaContext) (err error) {
 		return err
 	}
 
-	team.notify(mctx.Ctx(), keybase1.TeamChangeSet{MembershipChanged: true})
+	team.notify(mctx.Ctx(), keybase1.TeamChangeSet{MembershipChanged: true}, nextSeqno)
```
Comment (mlsteele) [inline go/teams/transactions.go]: nextSeqno-1 ? next vs last

File: go/teams/implicit.go [hunk 3]
```diff
@@ -176,7 +176,7 @@ func lookupImplicitTeamAndConflicts(ctx context.Context, g *libkb.GlobalContext,
 	team, err = Load(ctx, g, keybase1.LoadTeamArg{
 		ID:          teamID,
 		Public:      impTeamName.IsPublic,
-		ForceRepoll: true,
+		ForceRepoll: false, // it's important to hit the cache here, since we call this once per chat send!
```
Comment (mlsteele) [inline go/teams/implicit.go]: This function drives public functions which I'd expect to return the latest answer. Does chat not have its own cache?

Comment (mlsteele) [reply go/teams/implicit.go]: Or could plumb forceRepoll through

File: go/teams/transactions.go (hunk 2 above)
Comment (maxtaco) [reply go/teams/transactions.go]: ah good call, thanks. this was the particularly buggy bit.

Comment (maxtaco) [reply go/teams/transactions.go]: ...since it could result in stale values being used...

File: go/teams/implicit.go (hunk 3 above)
Comment (maxtaco) [reply go/teams/implicit.go]: no, chat just spams this (cc @mmaxim).  I was thinking though it should be OK to get a slightly stale answer here...

**Pull request merged**